includes/class-wc-google-analytics.php

```diff
@@ -705,4 +698,22 @@ public function add_wc_setup_task() {
 		);
 
 	}
+
+	/**
+	 * Add async to script tags with defined handles.
+	 *
+	 * @see https://github.com/woocommerce/woocommerce-google-analytics-integration/issues/186
+	 *
+	 * @param string $tag HTML for the script tag.
+	 * @param string $handle Handle of script.
+	 * @param string $src Src of script.
+	 *
+	 * @return string
+	 */
+	public function async_script_loader_tag( $tag, $handle, $src ) {
```

Comment:
It makes sense to prevent the header script from being blocking by adding async, but I'm just curious if that has any side effects.

> `async` : A script that will be downloaded in parallel to parsing the page, and executed as soon as it is available.

Is there a scenario where the code to trigger the tracking will be executed earlier (if the script takes longer to download)? The rest of the code is loaded through inline script where `async` doesn't seem like it would make sense (or have any effect).

I think we also might need a few more checks to make sure the script doesn't have an empty src or async has already been added, because I'm seeing this as the snippet (notice the double `async` for the gtag script):
```
<script async async src='https://www.googletagmanager.com/gtag/js?id=G-NGBB258&#038;ver=1.6.2' id='google-tag-manager-js'></script>
<script async id='google-tag-manager-js-after'>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    ...
```

Reply:
Seems like it's recommended by Google to add async: https://developers.google.com/tag-platform/gtagjs/install
But only on the first snippet. So maybe we need:

```php
		return str_replace( '<script src', '<script async src', $tag );
```

Reply:
Just adding a link to what the function looks like now in the blocks implementation: https://github.com/woocommerce/woocommerce-blocks/blob/v9.8.1/src/Domain/Services/GoogleAnalytics.php#L106-L115